### PR TITLE
Sequential min gru

### PR DESCRIPTION
**Pull request by lbaierreinio**
- Add sequential mode to the MinGRU model

The sequential mode is entirely implemented within the MinGRU class itself to avoid changes needing to be made to other files (e.g BiMinGru, MinGRUBlock, etc). Essentially the synthetic mode iterates over the sequence length dimension of a batch, updating the hidden states & masking where appropriate. This is based on https://d2l.ai/chapter_recurrent-modern/gru.html.

This implementation has the greatest flexibility because it means we can train or make predictions in either mode

### CODE REVIEW

File: src/layers/rnn/MinGRU.py [hunk 1]
```diff
@@ -53,7 +55,7 @@ def log_g(self, x):
         """
         return torch.where(x >= 0, torch.log(F.relu(x)+0.5), -F.softplus(-x))
 
-    def forward(self, x, *, h_prev=None, mask=None):
+    def forward(self, x, *, mask=None, is_sequential = False):
         """
         Compute the forward pass. Note that if h_prev is not none,
```
Comment (dlrudwo1269) [inline src/layers/rnn/MinGRU.py:60]: Can we update the docstring here since there has been some changes since we first wrote this?

File: src/layers/rnn/MinGRU.py [hunk 2]
```diff
@@ -67,30 +69,42 @@ def forward(self, x, *, h_prev=None, mask=None):
...
+            h = torch.zeros(batch_size, 0, self.dim_hidden)
+
+            for t in range(seq_len): # Iterate over sequence length
+                k_t = k[:, t, :]
+                tilde_h_t = tilde_h[:, t, :]
+                z_t = torch.sigmoid(k_t)
+                tilde_h_t_p = self.g(tilde_h_t)
+                h_prev = ((1 - z_t) * h_prev) + (z_t * tilde_h_t_p)
+                if mask is not None:
+                    mask_t = mask.squeeze(-1)[:, t]
```
Comment (dlrudwo1269) [inline src/layers/rnn/MinGRU.py:95]: Don't think we need `.squeeze` here since mask is [batch_size, seq_len]

**Review by dlrudwo1269 — DISMISSED** [review]
LGTM

File: src/layers/rnn/MinGRU.py (hunk 2 above)
Comment (lbaierreinio) [reply src/layers/rnn/MinGRU.py:95]: I believe we do because I unsqueeze before either sequential or parallel mode begins

Comment (dlrudwo1269) [reply src/layers/rnn/MinGRU.py:95]: Ah didn't catch that!
